### PR TITLE
add exception for dicts when evaluating sub in Cloudformation

### LINKED ISSUE #3068: find_all_interpolations crashing cloudformation with UserData
**Describe the issue**
Explain what you expected to happen when checkov crashed.
https://github.com/bridgecrewio/checkov/blob/master/checkov/cloudformation/graph_builder/utils.py#L38
this Line Crashes, when parsing cloudformation templates where the UserData is a dict
the function appears to expect a string

`def find_all_interpolations(str_value: str) -> List[str]:
    return re.findall(FIND_INTERPOLATION_PATTERN, str_value)
`

**Source code screenshot**
source_template
source_referenece

**Examples**
images provided
trace_console
debug
trace

**Exception Trace**

**Desktop (please complete the following information):**
 - OS: MAC
 - Checkov Version [2.0.1204] tried newer and older versions seems to remain

**Additional context**

### PR DESCRIPTION
**Pull request by maxamel**
This PR prevents crashing when processing Cloudformation templates with Sub functions containing dicts (currently similar to lists). 
 

Fixes #3068 

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by nimrodkor — COMMENTED** [review]
@maxamel Can you add a test case so we make sure we won't have a regression?

File: checkov/cloudformation/graph_builder/variable_rendering/renderer.py [hunk 1]
```diff
@@ -285,8 +285,8 @@ def _evaluate_getatt_connection(value: List[str], dest_vertex_attributes: Dict[s
 
     def _evaluate_sub_connection(self, value: str, dest_vertex_attributes: Dict[str, Any]) -> (
             Optional[str], Optional[str]):
-        if isinstance(value, list):
-            # TODO: Render values of list type
+        if isinstance(value, list) or isinstance(value, dict):
```
Comment (gruebel) [inline checkov/cloudformation/graph_builder/variable_rendering/renderer.py:288]:
```suggestion
        if isinstance(value, (list, dict)):
```
more beautiful 😄

**Comment by maxamel** [comment]
Added test containing Cloudformation template with a Sub function referencing a dict. This fix ensures we don't crash in this case.

**Review by nimrodkor — APPROVED** [review]
Just fix the UTs :)

**Pull request merged**